src/widget/wtracktableview.cpp

```diff
@@ -844,6 +844,44 @@ void WTrackTableView::keyPressEvent(QKeyEvent* event) {
     }
 }
 
+void WTrackTableView::resizeEvent(QResizeEvent* event) {
+    // When the tracks view shrinks in height, e.g. when other skin regions expand,
+    // and if the row was visible before resizing, scroll to it afterwards.
+
+    // these heights are the actual inner region without header, scrollbars and padding
+    int oldHeight = event->oldSize().height();
+    int newHeight = event->size().height();
+    bool maybeScrollTo = newHeight < oldHeight;
+
+    QModelIndex currIndex;
+    int rHeight = rowHeight(0);
+    if (maybeScrollTo) {
+        currIndex = currentIndex();
+        if (currIndex.isValid() && rHeight > 0) {
```

Comment:
```suggestion
    if (!maybeScrollTo) {
       QTableView::resizeEvent(event);
       return
    }
    
    QModelIndex currIndex = currentIndex();
    int rHeight = rowHeight(0);
    if (!currIndex.isValid() || rHeight <= 0) {        
       QTableView::resizeEvent(event);
       return
    }
```
Such eary returns would make this code easier to read.

Reply:
Alright, I reworked the entire resizeEvent to return early. I hope it's okay that I force-pushed, that made more sense to me than an undo/redo commit.